x/attribute/keeper/msg_server.go

```diff
@@ -66,6 +66,60 @@ func (k msgServer) AddAttribute(goCtx context.Context, msg *types.MsgAddAttribut
 	return &types.MsgAddAttributeResponse{}, nil
 }
 
+func (k msgServer) UpdateAttribute(goCtx context.Context, msg *types.MsgUpdateAttributeRequest) (*types.MsgUpdateAttributeResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	originalAttribute := types.Attribute{
+		Address:       msg.Account,
+		Name:          msg.Name,
+		AttributeType: msg.OriginalAttributeType,
+		Value:         msg.OriginalValue,
+	}
+
+	updateAttribute := types.Attribute{
+		Address:       msg.Account,
+		Name:          msg.Name,
+		AttributeType: msg.UpdateAttributeType,
+		Value:         msg.UpdateValue,
+	}
+
+	ownerAddr, err := sdk.AccAddressFromBech32(msg.Owner)
+	if err != nil {
+		return nil, err
+	}
+
+	err = k.Keeper.UpdateAttribute(ctx, originalAttribute, updateAttribute, ownerAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
+		telemetry.IncrCounterWithLabels(
+			[]string{types.ModuleName, types.EventTelemetryKeyAdd},
+			1,
+			[]metrics.Label{
+				telemetry.NewLabel(types.EventTelemetryLabelName, msg.Name),
+				telemetry.NewLabel(types.EventTelemetryLabelValue, string(msg.OriginalValue)),
+				telemetry.NewLabel(types.EventTelemetryLabelType, msg.OriginalAttributeType.String()),
+				telemetry.NewLabel(types.EventTelemetryLabelValue, string(msg.UpdateValue)),
+				telemetry.NewLabel(types.EventTelemetryLabelType, msg.UpdateAttributeType.String()),
+				telemetry.NewLabel(types.EventTelemetryLabelAccount, msg.Account),
+				telemetry.NewLabel(types.EventTelemetryLabelOwner, msg.Owner),
+			},
+		)
+	}()
+
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			types.EventTelemetryKeyUpdate,
```

Comment:
This should be `types. EventTypeAttributeUpdated` (this one might not yet exist).

Reply:
Thanks, delete distinct didn't exist either.  So I added that.